### PR TITLE
Client and cargo version update

### PR DESCRIPTION
**Pull request by dg-concordium**
## Purpose

Inform users about updates to client and cargo.

## Changes

Release notes and download links updated.

## Checklist

- [X] My code follows the style of this project.
- [X] The code compiles without warnings.
- [X] I have performed a self-review of the changes.
- [ ] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [ ] (If necessary) I have updated the CHANGELOG.

## CLA acceptance

_Remove if not applicable.

By submitting the contribution I accept the terms and conditions of the
Contributor License Agreement v1.0
- link: https://developers.concordium.com/CLAs/Contributor-License-Agreement-v1.0.pdf

- [X] I accept the above linked CLA.

### CODE REVIEW

**Comment by dg-concordium** [comment]
Check that links work and update checksums once packages are built.

File: source/mainnet/net/installation/downloads.rst [hunk 1]
```diff
@@ -80,28 +80,28 @@ Concordium Ledger App
...
 ========================
 
--  `Download the Mainnet Concordium Client for Linux <https://distribution.concordium.software/tools/linux/concordium-client_4.0.3-0>`_
+-  `Download the Mainnet Concordium Client for Linux <https://distribution.concordium.software/tools/linux/concordium-client_4.0.4-0>`_
 
       - SHA256 checksum of the download: :substitution-code:`|client-linux-checksum|`
       - :ref:`Verification instructions <verification-client-linux>`
 
--  `Download the Mainnet Concordium Client for macOS <https://distribution.concordium.software/tools/macos/signed/concordium-client_4.0.3-0.zip>`_
+-  `Download the Mainnet Concordium Client for macOS <https://distribution.concordium.software/tools/macos/signed/concordium-client_4.0.4-0.zip>`_
```
Comment (abizjak) [inline source/mainnet/net/installation/downloads.rst:91]: Link to be checked.

File: source/mainnet/net/installation/downloads-testnet.rst [hunk 2]
```diff
@@ -76,27 +76,27 @@ The version of the Ledger App is the same for Mainnet and Testnet. So if you alr
...
+-  `Download the Testnet Concordium Client for Windows <https://distribution.concordium.software/tools/windows/signed/concordium-client_4.0.4-0.exe>`_
 
-Cargo-concordium v2.0.1
+Cargo-concordium v2.0.3
 =======================
 
 Download cargo-concordium:
 
-   -  `Download Testnet cargo-concordium for Linux <https://distribution.concordium.software/tools/linux/cargo-concordium_2.0.1-0>`_
+   -  `Download Testnet cargo-concordium for Linux <https://distribution.concordium.software/tools/linux/cargo-concordium_2.0.3-0>`_
```
Comment (abizjak) [inline source/mainnet/net/installation/downloads-testnet.rst:95]:
```suggestion
   -  `Download Testnet cargo-concordium for Linux <https://distribution.concordium.software/tools/linux/cargo-concordium_2.0.2-0>`_
```

File: source/mainnet/net/installation/downloads-testnet.rst [hunk 3]
```diff
@@ -76,27 +76,27 @@ The version of the Ledger App is the same for Mainnet and Testnet. So if you alr
...
 
-   -  `Download Testnet cargo-concordium for MacOS <https://distribution.concordium.software/tools/macos/cargo-concordium_2.0.1-0>`_
+   -  `Download Testnet cargo-concordium for MacOS <https://distribution.concordium.software/tools/macos/cargo-concordium_2.0.3-0>`_
```
Comment (abizjak) [inline source/mainnet/net/installation/downloads-testnet.rst:97]:
```suggestion
   -  `Download Testnet cargo-concordium for MacOS <https://distribution.concordium.software/tools/macos/cargo-concordium_2.0.2-0>`_
```

File: source/mainnet/net/installation/downloads-testnet.rst [hunk 4]
```diff
@@ -76,27 +76,27 @@ The version of the Ledger App is the same for Mainnet and Testnet. So if you alr
...
 
-   -  `Download Testnet cargo-concordium for Windows <https://distribution.concordium.software/tools/windows/cargo-concordium_2.0.1-0.exe>`_
+   -  `Download Testnet cargo-concordium for Windows <https://distribution.concordium.software/tools/windows/cargo-concordium_2.0.3-0.exe>`_
```
Comment (abizjak) [inline source/mainnet/net/installation/downloads-testnet.rst:99]:
```suggestion
   -  `Download Testnet cargo-concordium for Windows <https://distribution.concordium.software/tools/windows/cargo-concordium_2.0.2-0.exe>`_
```

File: source/mainnet/net/installation/downloads.rst [hunk 5]
```diff
@@ -80,28 +80,28 @@ Concordium Ledger App
...
 
--  `Download the Mainnet Concordium Client for Windows <https://distribution.concordium.software/tools/windows/signed/concordium-client_4.0.3-0.exe>`_
+-  `Download the Mainnet Concordium Client for Windows <https://distribution.concordium.software/tools/windows/signed/concordium-client_4.0.4-0.exe>`_
 
-Cargo-concordium v2.0.1
+Cargo-concordium v2.0.3
```
Comment (abizjak) [inline source/mainnet/net/installation/downloads.rst:95]:
```suggestion
Cargo-concordium v2.0.2
```

File: source/mainnet/net/installation/downloads.rst [hunk 6]
```diff
@@ -80,28 +80,28 @@ Concordium Ledger App
...
 =======================
 
 Download cargo-concordium:
 
-   -  `Download Mainnet cargo-concordium for Linux <https://distribution.concordium.software/tools/linux/cargo-concordium_2.0.1-0>`_
+   -  `Download Mainnet cargo-concordium for Linux <https://distribution.concordium.software/tools/linux/cargo-concordium_2.0.3-0>`_
```
Comment (abizjak) [inline source/mainnet/net/installation/downloads.rst:100]:
```suggestion
   -  `Download Mainnet cargo-concordium for Linux <https://distribution.concordium.software/tools/linux/cargo-concordium_2.0.2-0>`_
```

File: source/mainnet/net/installation/downloads.rst [hunk 7]
```diff
@@ -80,28 +80,28 @@ Concordium Ledger App
...
 
-   -  `Download Mainnet cargo-concordium for MacOS <https://distribution.concordium.software/tools/macos/cargo-concordium_2.0.1-0>`_
+   -  `Download Mainnet cargo-concordium for MacOS <https://distribution.concordium.software/tools/macos/cargo-concordium_2.0.3-0>`_
```
Comment (abizjak) [inline source/mainnet/net/installation/downloads.rst:102]:
```suggestion
   -  `Download Mainnet cargo-concordium for MacOS <https://distribution.concordium.software/tools/macos/cargo-concordium_2.0.2-0>`_
```

File: source/mainnet/net/installation/downloads.rst [hunk 8]
```diff
@@ -80,28 +80,28 @@ Concordium Ledger App
...
 
-   -  `Download Mainnet cargo-concordium for Windows <https://distribution.concordium.software/tools/windows/cargo-concordium_2.0.1-0.exe>`_
+   -  `Download Mainnet cargo-concordium for Windows <https://distribution.concordium.software/tools/windows/cargo-concordium_2.0.3-0.exe>`_
```
Comment (abizjak) [inline source/mainnet/net/installation/downloads.rst:104]:
```suggestion
   -  `Download Mainnet cargo-concordium for Windows <https://distribution.concordium.software/tools/windows/cargo-concordium_2.0.2-0.exe>`_
```

File: source/mainnet/net/installation/downloads.rst (hunk 1 above)
Comment (abizjak) [inline source/mainnet/net/installation/downloads.rst:91]:
```suggestion
-  `Download the Mainnet Concordium Client for macOS <https://distribution.concordium.software/tools/macos/signed/concordium-client-4.0.4-0.pkg>`_
```

File: source/mainnet/net/installation/downloads-testnet.rst [hunk 9]
```diff
@@ -76,27 +76,27 @@ The version of the Ledger App is the same for Mainnet and Testnet. So if you alr
...
+   - SHA256 checksum of the download: ``8d4bc3e630c958d0443fa5cdbff2bcfba3bca30424346fd8ad6f5415aa52ab95``
 
--  `Download the Testnet Concordium Client for macOS <https://distribution.concordium.software/tools/macos/signed/concordium-client_4.0.3-0.zip>`_
+-  `Download the Testnet Concordium Client for macOS <https://distribution.concordium.software/tools/macos/signed/concordium-client-4.0.4-0.pkg>`_
 
--  `Download the Testnet Concordium Client for Windows <https://distribution.concordium.software/tools/windows/signed/concordium-client_4.0.3-0.exe>`_
+-  `Download the Testnet Concordium Client for Windows <https://distribution.concordium.software/tools/windows/signed/concordium-client_4.0.4-0.exe>`_
 
-Cargo-concordium v2.0.1
+Cargo-concordium v2.0.3
```
Comment (DOBEN) [inline source/mainnet/net/installation/downloads-testnet.rst:92]: should this be version v2.0.2?

Comment (dg-concordium) [reply source/mainnet/net/installation/downloads-testnet.rst]: Fixed.

**Review by abizjak — APPROVED** [review]

**Pull request merged**